airflow-core/src/airflow/api_fastapi/core_api/security.py

```diff
@@ -329,14 +329,23 @@ def _requires_access(
         raise HTTPException(status.HTTP_403_FORBIDDEN, "Forbidden")
 
 
-def is_safe_url(target_url: str) -> bool:
+def is_safe_url(target_url: str, request: Request | None = None) -> bool:
     """
     Check that the URL is safe.
 
     Needs to belong to the same domain as base_url, use HTTP or HTTPS (no JavaScript/data schemes),
     is a valid normalized path.
     """
-    base_url = conf.get("api", "base_url")
+    base_url: str | None = conf.get("api", "base_url")
```

Comment:
The way the config is defined it's impossible for this to ever actually be None btw -- it's got a default value, so the only way of overriding it would be to an empty string

Reply:
Even with your change, it still has a default of empty string :)

Reply:
yes but if it's empty, we then take it from the `referer` header of the request. Which might be missing...

But i agree this is confusing and need to be cleaned so it's clearer.